test/metabase/models/action_test.clj

```diff
@@ -26,15 +27,20 @@
 (deftest hydrate-implicit-action-test
   (mt/test-drivers (mt/normal-drivers-with-feature :actions/custom)
     (mt/with-actions-test-data-and-actions-enabled
-      (mt/with-actions [{:keys [action-id] :as _context} {:type :implicit}]
-        (is (partial= {:id          action-id
-                       :name        "Update Example"
-                       :database_id (mt/id)
-                       :parameters  [(if (= driver/*driver* :h2)
-                                       {:type :type/BigInteger}
-                                       {:type :type/Integer})
-                                     {:type :type/Text, :id "name"}]}
-                      (action/select-action :id action-id))))))
+      (let [query (mt/mbql-query categories)]
+        (mt/with-actions [_model {:dataset         true
+                                  :dataset_query   query
+                                  :result_metadata (assoc-in (get-in (qp/process-query query) [:data :results_metadata :columns])
+                                                             [1 :display_name] "Display Name")}
+                          {:keys [action-id] :as _context} {:type :implicit}]
+          (is (partial= {:id          action-id
+                         :name        "Update Example"
+                         :database_id (mt/id)
+                         :parameters  [(if (= driver/*driver* :h2)
+                                         {:type :type/BigInteger}
+                                         {:type :type/Integer})
+                                       {:type :type/Text, :id "name" :display-name "Display Name"}]}
+                        (action/select-action :id action-id)))))))
```

Comment:
Not the strongest test since it pokes the `:display_name` in manually and then effectively checks that it's there. You might be able to do something clever with `qp/query->expected-cols` but I don't have a great suggestion off the top of my head.

Reply:
I think that's basically all that needs testing though. That the the `display_name` is pulled from the `result_metadata`. I can't imagine another way. The `display_name` in `result_metadata` is set by the frontend, so associng it in like this is effectively simulating that.